mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java

```diff
@@ -355,6 +371,15 @@ private void processIntent(Intent intent) {
                 CloudMessagingHelper.onNotificationSelected(this, intent);
                 onNotificationSelected(intent);
                 break;
+            case ACTION_HABPANEL_SELECTED:
+                mShouldStartHabpanel = true;
+                openHabpanelIfNeeded();
+                break;
+            case ACTION_VOICE_RECOGNITION_SELECTED:
+                if (mServerProperties != null) {
+                    launchVoiceRecognition();
```

Comment:
This should be wrapped in a `launchVoiceRecognitionIfNeeded()` function and be called after the serverprops have been fetched.